syntax = "proto3";

import "is/msgs/validate.proto";
import "is/msgs/camera.proto";

enum CameraDrivers {
  NOT_SPECIFIED = 0;
  FLYCAPTURE = 1; 
  SPINNAKER = 2;
}

message CameraGatewayOptions {
  string broker_uri = 1;
  string zipkin_host = 2;
  uint32 zipkin_port = 3;
  string camera_ip = 4;
  int32 camera_id = 5 [(is.validate.rules).int32 = {gte: 0}];
  int32 packet_delay = 6 [(is.validate.rules).int32 = {gte: 0, lte: 6250}];
  int32 packet_size = 7 [(is.validate.rules).int32 = {gte: 576, lte: 9000}];
  bool reverse_x = 8;
  bool reverse_y = 9;
  int32 parallelism = 10;
  is.vision.CameraConfig initial_config = 11;
  CameraDrivers camera_driver = 12;
}